src/mammos_entity/io.py

```diff
@@ -415,20 +440,42 @@ def _represent_string(dumper, value):
 
 
 class EntityCollection:
-    """Container class storing entity-like objects."""
+    """Container class storing entity-like objects.
+
+    Attributes:
+        description: String containing information about the ``EntityCollection``.
+    """
+
+    def __init__(self, description: str | None = None, **kwargs):
+        """Initialize EntityCollection, keywords become attributes of the class.
 
-    def __init__(self, **kwargs):
-        """Initialize EntityCollection, keywords become attributes of the class."""
+        Args:
+            description: Information string to assign to ``description`` attribute.
+            **kwargs : entities to be stored in the collection.
+        """
+        self.description = description if description is not None else ""
```

Comment:
```suggestion
        self.description = description
```

Please make description a property.